src/merge.ts

```diff
@@ -0,0 +1,44 @@
+import { EventPayload } from './EventPayload';
+
+/**
+ * Used for merging request JSON into an existing EventPayload.
+ * Returns a new EventPayload, with the original object unaffected.
+ * The merge occurs using the following conventions:
+ *
+ *  Merging an existing key with null/undefined deletes the key.
+ *  Merging an existing key with non-null value updates the value.
+ *  Merging a non-existing key adds the key/value.
+ *  When the value is an object, the above is applied iteratively.
+ */
+export const merge = (
+  original: EventPayload,
+  newValues: Partial<Record<keyof EventPayload, unknown>>
+): EventPayload => {
+  const copy: Record<string, unknown> = { ...original };
+  const stack: Array<[Record<string, unknown>,
```

Comment:
nit: don't need to state the type. It's inferable from the assignment.

Reply:
Small correction, if you want to keep it as and Array<tuple>, you need to do something like the below
`const stack = new Array([copy, newValues])`

Reply:
As discussed offline, this doesn't work as the inferred type is not exactly the same.

Reply:
Going with `const stack: Array<[PartialPayload, PartialPayload]> = [[copy, newValues]];` as discussed offline